### PR TITLE
Updating README, meta #140

### PR DESCRIPTION
**Pull request by rorticus**
**Type:** feature

The following has been addressed in the PR:

* [x] There is a related issue
* [ ] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [ ] Unit or Functional tests are included in the PR

**Description:**

Updated README!

Resolves: https://github.com/dojo/meta/issues/140

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,17 +85,62 @@ to the key/value pair. This allows for the garbage collector to remove pairs.
...
+### Object
+
+The `dojo/shim/object` provides implementations of `Object` methods.
+
+* is
+* getOwnPropertySumbols
+* getOwnPropertyNames
+* getOwnPropertyDescriptor
+* values
+* entires
```
Comment (dylans) [inline README.md]: typo

File: README.md [hunk 2]
```diff
@@ -35,17 +85,62 @@ to the key/value pair. This allows for the garbage collector to remove pairs.
...
+* `isFinite`
+* `isInteger`
+* `isSafeInteger`
+
+### Object
+
+The `dojo/shim/object` provides implementations of `Object` methods.
+
+* is
+* getOwnPropertySumbols
```
Comment (dylans) [inline README.md]: typo

File: README.md [hunk 3]
```diff
@@ -35,17 +85,62 @@ to the key/value pair. This allows for the garbage collector to remove pairs.
...
+
+[`@dojo/shim/Promise`](docs/Promise.md) is an implementation of the [ES2015 Promise specification](http://www.ecma-international.org/ecma-262/6.0/#sec-promise-objects).
+
+### String
+
+The [`@dojo/shim/string`](docs/string.md) module contains `String` methods.
+
+### Symbols
+
+`@dojo/shim/Symbol` provides an implementation of the [ES2015 Symbol specification](http://www.ecma-international.org/ecma-262/6.0/#sec-symbol-objects) for browsers that do not natively support `Symbol`.
```
Comment (dylans) [inline README.md]: just browsers, or JS environments? (Not that I know of a non-browser JS engine we support that doesn't support Symbols), just don't think the shim is browser specific.

File: docs/Observable.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
+# Observable
+
+The `@dojo/shim/Observable` class is an implementation of the proposed [Observable specification](https://tc39.github.io/proposal-observable/).
+
+An `Observable` is used to simplify push-based data sources like events, timed events, or sockets.
+
+Here we create an Observable that publishes an event every 1 second:
+
+```typescript
+const timedObservable = new Observable(observer => {
```
Comment (dylans) [inline docs/Observable.md:10]: whitespace indentation on this example looks off, though that may be Github... should use tabs

File: docs/Observable.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+```
+
+Here we are creating an Observable that publishes an event every second. Note that one timer is started _per subscriber_, and if there are no subscribers, no timers are scheduled.
+
+## Creation
+
+Observables can be created with their constructor:
+
+```typescript
+new Observable((observer: SubscriptionObserver) => {
```
Comment (dylans) [inline docs/Observable.md:43]: tabs over spaces

File: docs/Observable.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+Observables can also be created with the `of` and `from` static methods.
+
+When creating an observable with a constructor, you receive a single argument, the `SubscriptionObserver`. Interactions between the Observable and the subscriber are handled through this object.
+
+The `SubscriptionObserver` can publish values, errors, or signal completion of the Observable.
+
+```typescript
+// publish two values, then signal completion
+new Observable(observer => {
```
Comment (dylans) [inline docs/Observable.md:59]: tabs over spaces

File: docs/Observable.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+  observer.next(1);
+  observer.next(2);
+  observer.complete();
+});
+
+// publish a value, then error
+new Observable(observer => {
```
Comment (dylans) [inline docs/Observable.md]: tabs over spaces

File: docs/Observable.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
...
+
+### `of`
+
+`Observable.of` is used to create an Observable from a series of parameters. The observable will return each parameter in order.
+
+```typescript
+// Observable.of
+const obs = Observable.of(1, 2, 3);
+obs.subscribe(value => {
+  console.log(value);
```
Comment (dylans) [inline docs/Observable.md]: tabs over spaces

File: docs/Observable.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+
+### `from`
+
+`Observable.from` is used to create an Observable from a list of values, either an array, an `Iterable` object, or another Observable. The subscribers will be called once for every value in the wrapped object.
+
+```typescript
+// Observable.from
+const obs = Observable.from(someIterator);
+obs.subscribe(value => {
+  // handle iterator value
```
Comment (dylans) [inline docs/Observable.md]: tabs over spaces

File: docs/Observable.md [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+
+When the `subscribe` method is called on an Observable, a `Subscription` is created. This `Subscription` can be used to check if the subscription is still active, or to unsubscribe from the Observable.
+
+```typescript
+const subscription = someObservable.subscribe(someSubscriber);
+
+// ...
+
+if (subscription.closed) {
+  // handle a closed subscription
```
Comment (dylans) [inline docs/Observable.md]: tabs over spaces

File: docs/Observable.md [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+} else {
+  // unsubscribe from the subscription
```
Comment (dylans) [inline docs/Observable.md]: tabs over spaces

File: docs/Observable.md [hunk 12]
```diff
@@ -0,0 +1,150 @@
...
+```
+
+Once you are unsubscribed, you will no longer received published values.
+
+The `subscribe` method takes either a list of function handlers or an `Observer` which contains the handlers.
+
+```typescript
+// note that the error handler and completion handler are optional
+someObservable.subscribe((value) => {
+  // next
```
Comment (dylans) [inline docs/Observable.md]: tabs over spaces

File: docs/Promise.md [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+
+Promises are used to reduce complications of asynchronous callbacks.
+
+## Creating Promises
+
+You can create a promise using the constructor:
+
+```typescript
+const p = new Promise((resolve, reject) => {
+  // some long running task
```
Comment (dylans) [inline docs/Promise.md]: tabs over spaces

File: docs/Promise.md [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+```typescript
+const resolvePromise = Promise.resolve();
+const rejectedPromise = Promise.reject();
+```
+
+Values can be wrapped in a resolved promise to facilitate things
+like simple caching:
+
+```typescript
+function loadData(): Promise<Data> {
```
Comment (dylans) [inline docs/Promise.md]: tabs over spaces

File: docs/Promise.md [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+
+or by using static convenience methods:
+
+```typescript
+const resolvePromise = Promise.resolve();
+const rejectedPromise = Promise.reject();
+```
+
+Values can be wrapped in a resolved promise to facilitate things
+like simple caching:
```
Comment (dylans) [inline docs/Promise.md]: Don't like "things like..." perhaps to facilitate simple caching and other <insert word that describes the right pattern>

File: docs/Promise.md [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+const task2 = startLongRunningTask();
+
+Promise.race([task1, task2]).then(() => {
+  // either task1 or task2 has completed
+});
+```
+
+## Using Promises
+
+You interact with promises using the `then` and `catch` methods.
```
Comment (dylans) [inline docs/Promise.md]: Weird that we changed to "You" here.

File: docs/Promise.md [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+
+### Then
+
+`.then(successHandler, errorHandler)` is used to handle the resolution (or failure) of a promise. `errorHandler` is optional. The resolve, or rejected, value is passed to the handler.
+
+```typescript
+const resolvePromise = Promise.resolve(true);
+const rejectedPromise = PRomise.reject(new Error('some error'));
```
Comment (dylans) [inline docs/Promise.md]: typo

File: docs/Promise.md [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+
+resolvedPromise.then((value) => {
+  // value is true
```
Comment (dylans) [inline docs/Promise.md]: tabs over spaces

**Review by dylans — CHANGES_REQUESTED** [review]
Very good, just a lot of little details and then it's done.

**Comment by rorticus** [comment]
@dylans all the requested updates have been made!

**Review by dylans — APPROVED** [review]

**Pull request merged**